projects/components/src/table/table-csv-downloader.service.ts

```diff
@@ -0,0 +1,72 @@
+import { EMPTY, Observable, of, Subject } from 'rxjs';
+import { Injectable } from '@angular/core';
+import { TableColumnConfigExtended } from './table.service';
+import { TableRow } from './table-api';
+import { FileDownloadService } from '../download-file/service/file-download.service';
+import { map, switchMap } from 'rxjs/operators';
+import { isEmpty, isNil, isString } from 'lodash-es';
+import { Dictionary } from '@hypertrace/common';
+import { NotificationService } from '../notification/notification.service';
+
+@Injectable({ providedIn: 'root' })
+export class TableCsvDownloaderService {
+  private readonly csvDownloadRequestSubject: Subject<string> = new Subject<string>();
+  public csvDownloadRequest$: Observable<string> = this.csvDownloadRequestSubject.asObservable();
+
+  public constructor(
+    private readonly fileDownloadService: FileDownloadService,
+    private readonly notificationService: NotificationService
+  ) {}
+  public triggerDownload(tableId: string): void {
+    this.csvDownloadRequestSubject.next(tableId);
+  }
+
+  public executeDownload(
```

Comment:
@arjunlalb We show a confirmation modal before downloading csv. In a follow-up pr, do you think we should add that logic inside this method?

Reply:
@anandtiwary we could.